plugins/rule_engines/irods_rule_language/src/configuration.cpp

```diff
@@ -443,13 +480,18 @@ int load_rules(const char* irbSet, const std::vector<std::string> &irbs, const i
                     getSystemFunctions( ruleEngineConfig.sysFuncDescIndex->current, ruleEngineConfig.sysRegion );
                 }
 
-                make_copy copy_rule_base_files(irbs, pid);
+                in_memory_rulebases copy_rule_base_files(irbs);
                 for(auto const & irb: irbs) {
-                    int i = readRuleStructAndRuleSetFromFile( irb.c_str(), get_rule_base_path_copy(irb, pid).c_str() );
+                    try {
+                        auto* rule_ptr = const_cast<char*>(&copy_rule_base_files.get_rulebase(irb)[0]);
+                        int i = readRuleStructAndRuleSetFromBuffer(irb.c_str(), rule_ptr);
 
-                    if ( i != 0 ) {
-                        ruleEngineConfig.ruleEngineStatus = INITIALIZED;
-                        return i;
+                        if ( i != 0 ) {
+                            ruleEngineConfig.ruleEngineStatus = INITIALIZED;
+                            return i;
+                        }
+                    } catch(std::out_of_range& ex) {
+                        logger::rule_engine::error("load_rules: attempted to access nonexistent rulebase [{}]", irb);
```

Comment:
Same question here about whether this is serious enough to return early with an error. Maybe the answer is no.